csrc/entrypoint.cpp

```diff
@@ -11,6 +13,9 @@ struct ThreadLocalConfig {
 };
 static thread_local ThreadLocalConfig thread_local_config;
 
+static std::string last_error_message;
```

Comment:
(same for all these - would be great if we skip error handling to keep code minimal)